spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceBucket.java

```diff
@@ -0,0 +1,163 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.WritableResource;
+import org.springframework.util.Assert;
+
+/**
+ * Implements {@link WritableResource} for buckets in Google Cloud Storage.
+ */
+public class GoogleStorageResourceBucket implements WritableResource {
+
+	private final Storage storage;
+
+	private final String bucketName;
+
+	public GoogleStorageResourceBucket(Storage storage, String bucketName,
+			boolean createBucketIfNotExists) {
+		Assert.notNull(storage, "Storage object can not be null");
+		Assert.notNull(bucketName, "Bucket name can not be null");
+
+		this.storage = storage;
+		this.bucketName = bucketName;
+
+		if (!exists() && createBucketIfNotExists) {
+			this.storage.create(BucketInfo.newBuilder(this.bucketName).build());
+		}
+	}
+
+	@Override
+	public boolean exists() {
+		try {
+			return throwExceptionForNullBucket(resolve()).exists();
```

Comment:
Again, I know this is copy&pasted, but can we change this to call `resolve()` and check if the result is `null`?
That would spare us the unnecessary and costly exception handling.

Reply:
Will do.

Reply:
Oh on second look, I had this here because resolve() might return null, and calling .exists() would then give an NPE.